src/CodeGen.cpp

```diff
@@ -174,7 +174,7 @@ bool CodeGen::llvm_NVPTX_enabled = false;
 bool CodeGen::llvm_Mips_enabled = false;
 
 void CodeGen::compile(Stmt stmt, string name,
-                      const vector<Argument> &args,
+                      const ArgInfo &arg_info,
```

Comment:
Conceptually, these two fields really are a single argument; it makes it clear that "num_outputs" is tied intimately to the vector of Arguments, rather than a separate argument. It also makes "pass-thru" calls easier.

Re: the merge with other branches, adding an argument to compile() rather than simply changing it seems like it's going to be a comparable amount of work either way. 

(Side note: haven't those branches landed yet? When are they landing?)